src/DOMParser.ts

```diff
@@ -163,6 +163,20 @@ export class DOMParser {
       current.appendChild(doc.createComment(comment));
     };
 
+    //XML needs to start with <...(root element) or <?...(prolog) but not a comment <!-...
+    if (
+      source != null &&
+      (source.startsWith('<') !== true || source.startsWith('<!-') === true)
```

Comment:
Can be simplified to `(!source.startsWith('<') || source.startsWith('<!--'))`. (Also note that a comment needs two dashes)

Reply:
Shortened and fixed in [51cdae6](https://github.com/journeyapps/domparser.js/pull/22#:~:text=%E2%80%A6-,51cdae6,-Changed%20test%20description)